hs_core/management/commands/list_published_by_year.py

```diff
@@ -37,47 +40,74 @@ def add_arguments(self, parser):
             help='limit to resources owned by specific user'
         )
 
+        parser.add_argument(
+            '--csv',
+            action='store_true',  # True for presence, False for absence
+            dest='csv',
+            help='export as csv'
+        )
+
     def handle(self, *args, **options):
         days = options['days']
         resources = BaseResource.objects.filter(raccess__published=True)
         owner = options['owned_by']
         res_type = options['type']
+        csv = options['csv']
+        year = options['year']
+        file_name = ''
 
         if owner is not None:
             try:
                 owner = User.objects.get(username=owner)
                 resources.filter(r2urp__user=owner,
                                  r2urp__privilege=PrivilegeCodes.OWNER)
+                if csv:
+                    file_name += f"_owner={owner}"
             except ObjectDoesNotExist:
                 print(f"User matching {owner} not found")
 
         if res_type is not None:
             if res_type in ["CompositeResource", "CollectionResource"]:
                 resources.filter(resource_type=res_type)
+                file_name += f"_restype={res_type}"
             else:
                 print(f"Type {res_type} is not supported. Must be 'CompositeResource' or 'CollectionResource'")
 
+        if csv:
+            file_name += f"_run={timezone.now()}"
+            if year:
+                file_name += f"_year={year}"
+
         resources = resources.order_by(F('updated').asc(nulls_first=True))
+        site_url = hydroshare.utils.current_site_url()
+
+        if csv:
+            self.csvfile = open(f'published_resources{file_name}.csv', 'w')
```

Comment:
Wondering where this file will be created.

Reply:
modified so that the `--csv` arg now takes a directory to export into